hack/e2e.go

```diff
@@ -207,6 +207,11 @@ func run(deploy deployer) error {
 	}
 
 	if *up {
+		// If we tried to bring the Kubemark cluster up, make a courtesy
+		// attempt to bring it down so we're not leaving resources around.
+		if *down {
+			defer xmlWrap("TearDown", deploy.Down)
+		}
 		// Start the cluster using this version.
 		if err := xmlWrap("Up", deploy.Up); err != nil {
 			return fmt.Errorf("starting e2e cluster: %s", err)
```

Comment:
Sure, I didn't want to imply that you broke it on purpose.
I agree with the goal (always attempt to teardown exactly once). Though this isn't super trivial now, and I think that the issue this is solving is pretty serious (we just faced exactly the same for GCE - the lack of quota there is not configuration issue - it's just leaking resources because of not calling tear down after e.g. timeedout runs).

So WDYT about fixing it with this PR as is (to fix those leaks), and leave a TODO to fix it properly in the upcoming days?